lib/Controller/SlaveController.php

```diff
@@ -181,8 +218,40 @@ protected function decodeJwt($jwt) {
 
 		$uid = $decoded['uid'];
 		$password = $this->crypto->decrypt($decoded['password'], $key);
+		$options = isset($decoded['options']) ? $decoded['options'] : json_encode([]);
+
+		return [$uid, $password, json_decode($options, true)];
+	}
+
+
+	/**
+	 * create new user if the user doesn't exist yet on the client node
+	 *
+	 * @param string $uid
+	 * @param array $options
+	 * @return bool
+	 */
+	protected function autoprovisionIfNeeded($uid, $options) {
+
+		// make sure that a valid UID is given
+		if (empty($uid)) {
+			$this->logger->error('Uid "' . $uid . '" is not a valid uid', ['app' => $this->appName]);
+			throw new \InvalidArgumentException('No valid uid given. Given uid: ' . $uid);
+		}
+
+		$userExists = $this->userManager->userExists($uid);
+		if($userExists === true) {
+			$this->userBackend->updateAttributes($uid, $options);
+			return true;
+		}
+
+		if(!$userExists) {
```

Comment:
minor: can be simiplified, the previous if is not necessary, updateAttributes is then unconditional.